### DISCUSSION TITLE
How to have npm and node with dotnet-core buildpack

### DISCUSSION THREAD

**Discussion by sandrotaje** [paketo-buildpacks/feedback#36] [Q&A]
Hi,
Inside the [dotnet core repo](https://github.com/paketo-buildpacks/dotnet-core) it is stated that it's possible to use `NodeJs` but I can't find how to do it.

I tried to use the `BP_NODE_VERSION` environment variable and it correctly detects to use also the [node-engine](https://github.com/paketo-buildpacks/node-engine) buildpack but the `dotnet` buildpack doesn't have it. It seems somehow it misses the `Build Plan TOML`.

Is this a bug or I'm doing something wrong?

**Comment by ryanmoran** [comment]
Could you provide some more details about your application? If its possible to share the source code itself, or a similar app that reproduces the error, that would help immensely to resolve your issue.

It would also be great if you could include the build logs output by `pack --verbose build <app-name>`.

\cc @paketo-buildpacks/dotnet-core-maintainers

**Reply by sandrotaje** [reply]
Hi, unfortunatly I can't share the source code but reading the `dotnet-publish` code I think I found the problem. Right now into the `findInFile` function it is checked just the first `.csproj` file but in my use case I have another `.csproj` included with ProjectReference and that in turn includes the target part of the file from another file. 

I'm not a go developer but I will try to open a PR to add the recursive scanning feature to the `findInFile` function. 

What do you think?

**Reply by ryanmoran** [reply]
That sounds reasonable. You can also feel free to open an issue on the `dotnet-publish` buildpack repo explaining your problem.
